File: 434.NumberOfSegmentsInAString/segments.cc
class Solution {
public:
    int countSegments(string s) {
        int count = 0;
        int flag = 0;
        for (char c : s) {
            if (!isspace(c)) {
                flag = 1;
            }
            else {
                if (flag == 1) {
                    count++;
                }
                flag = 0;
            }
        }
        return count + flag;
    }
};

File: 434.NumberOfSegmentsInAString/segments.py
class Solution(object):
    def countSegments(self, s):
        """
        :type s: str
        :rtype: int
        """
        flag = 0
        count = 0
        for c in s:
            if not c.isspace():
                flag = 1
            else:
                if flag == 1:
                    count += 1
                flag = 0

        return count + (flag == 1)

if __name__ == '__main__':
    s = 'Hello, my name is John. '
    print Solution().countSegments(s)
